src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -245,6 +245,11 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             else if (policy.Origins.Contains(origin))
             {
                 result.AllowedOrigin = origin;
+
+                if(policy.Origins.Count > 1)
```

Comment:
I believe this is correct. Even if a single origin is returned, the `Vary: origin` header should be set on the response to allow HTTP caches to respond correctly. Chrome in particular will return cross-origin requests from cache if this header is not set. This causes errors, for example, if applications in separate tabs make CORS requests to the same resource. Chrome will cache the first request -- which contains the header `Access-Control-Allow-Origin: http://origin1` but not `Vary: origin` -- and serve the cached response up to the second tab, which will fail because it's origin doesn't match the `Access-Control-Allow-Origin` header.

For discussion see: https://bugs.chromium.org/p/chromium/issues/detail?id=409090

#https://github.com/IdentityModel/oidc-client-js/issues/275